parlai/mturk/core/react_server/dev/components/core_components.jsx

```diff
@@ -794,9 +857,11 @@ class DoneResponse extends React.Component {
       paddingRight: '25px',
       float: 'left',
     };
-    let done_button = <XDoneButton {...this.props} />;
-    if (!this.props.task_done) {
-      done_button = null;
+    let button = null;
+    if (this.props.task_done) {
+      button = <XDoneButton {...this.props} />;
+    } else if (this.props.subtask_done && this.props.show_next_task_button !== null && this.props.show_next_task_button) {
```

Comment:
Nit: There's no need to do the explicit !== null check considering you're already checking for truthiness of `show_next_task_button`.